app/assets/javascripts/discourse-common/addon/resolver.js

```diff
@@ -85,252 +66,255 @@ export function buildResolver(baseName) {
+        let match = Object.entries(this._normalizeCache).find(
+          ([key, value]) => {
+            return (
+              value === normalized &&
+              (wasCorrected || this._uncorrectedValues[key])
+            );
+          }
+        );
+        if (match) {
+          deprecated(
+            function () {
+              let message = `Both ${fullName} and ${match[0]} normalized to the same value: ${normalized}. In the future they will not normalize to the same thing.`;
+              if (this._uncorrectedValues[fullName]) {
+                message += ` ${fullName} will normalize to ${this._uncorrectedValues[fullName]}`;
+              }
+              if (this._uncorrectedValues[match[0]]) {
+                message += ` ${match[0]} will normalize to ${
+                  this._uncorrectedValues[match[0]]
+                }`;
+              }
+              return message;
+            }.call(this)
+          );
         }
       }
-      return module;
-    },
-
-    resolveWidget(parsedName) {
-      return this.customResolve(parsedName) || this._super(parsedName);
-    },
 
-    resolveAdapter(parsedName) {
-      return this.customResolve(parsedName) || this._super(parsedName);
-    },
+      return normalized;
+    }
 
-    resolveModel(parsedName) {
-      return this.customResolve(parsedName) || this._super(parsedName);
-    },
-
-    resolveView(parsedName) {
-      return this.customResolve(parsedName) || this._super(parsedName);
-    },
+    chooseModuleName(moduleName, parsedName) {
+      let resolved = super.chooseModuleName(moduleName, parsedName);
+      if (resolved) {
+        return resolved;
+      }
 
-    resolveHelper(parsedName) {
-      return (
-        findHelper(parsedName.fullNameWithoutType) ||
-        this.customResolve(parsedName) ||
-        this._super(parsedName)
-      );
-    },
+      const standard = parsedName.fullNameWithoutType;
 
-    resolveController(parsedName) {
-      return this.customResolve(parsedName) || this._super(parsedName);
-    },
+      let variants = [standard];
 
-    resolveComponent(parsedName) {
-      return this.customResolve(parsedName) || this._super(parsedName);
-    },
+      if (standard.includes("/")) {
+        variants.push(parsedName.fullNameWithoutType.replace(/\//g, "-"));
+      }
 
-    resolveService(parsedName) {
-      return this.customResolve(parsedName) || this._super(parsedName);
-    },
+      for (let name of variants) {
+        // If we end with the name we want, use it. This allows us to define components within plugins.
+        const suffix = parsedName.type + "s/" + name;
+        resolved = lookupModuleBySuffix(dasherize(suffix));
+        if (resolved) {
+          return resolved;
+        }
+      }
+    }
 
-    resolveRawView(parsedName) {
-      return this.customResolve(parsedName) || this._super(parsedName);
-    },
+    resolveHelper(parsedName) {
+      return findHelper(parsedName.fullNameWithoutType);
```

Comment:
No need for super, since ember-resolver calls `resolveOther` if the specific resolver doesn't handle it.